internal/worker/objectstore/worker.go

```diff
@@ -279,7 +280,7 @@ func (w *objectStoreWorker) initObjectStore(namespace string) error {
 		if namespace == database.ControllerNS {
 			metadataService = w.cfg.ControllerMetadataService
 		} else {
-			metadataService = w.cfg.ModelMetadataServiceGetter.ForModelUUID(namespace)
+			metadataService = w.cfg.ModelMetadataServiceGetter.ForModelUUID(model.UUID(namespace))
```

Comment:
We should probably validate this.

Reply:
@SimonRichardson I added a patch to do this, however it was causing the objectstore worker tests to freeze up and time out, so I had to back it out.
```diff
diff --git a/internal/worker/objectstore/worker.go b/internal/worker/objectstore/worker.go
index c015e5e1e7..5dbdba4cb0 100644
--- a/internal/worker/objectstore/worker.go
+++ b/internal/worker/objectstore/worker.go
@@ -280,7 +281,11 @@ func (w *objectStoreWorker) initObjectStore(namespace string) error {
 		if namespace == database.ControllerNS {
 			metadataService = w.cfg.ControllerMetadataService
 		} else {
-			metadataService = w.cfg.ModelMetadataServiceGetter.ForModelUUID(model.UUID(namespace))
+			modelID := model.UUID(namespace)
+			if err = modelID.Validate(); err != nil {
+				return nil, fmt.Errorf("invalid model ID %q: %w", namespace, err)
+			}
+			metadataService = w.cfg.ModelMetadataServiceGetter.ForModelUUID(modelID)
 		}
 
 		objectStore, err := w.cfg.NewObjectStoreWorker(
```